e2e/tests/functional/plugins/notebook/notebookWithCouchDB.e2e.spec.js

```diff
@@ -0,0 +1,182 @@
+
+/*
+This test suite is dedicated to tests which verify the basic operations surrounding Notebooks with CouchDB.
+*/
+
+const { test, expect } = require('../../../../baseFixtures');
+const { createDomainObjectWithDefaults } = require('../../../../appActions');
+
+test.describe('Notebook Network Request Inspection @couchdb', () => {
+    let testNotebook;
+    test.beforeEach(async ({ page }) => {
+        //Navigate to baseURL
+        await page.goto('./', { waitUntil: 'networkidle' });
+
+        // Create Notebook
+        testNotebook = await createDomainObjectWithDefaults(page, {
+            type: 'Notebook',
+            name: "TestNotebook"
+        });
+    });
+
+    test('Inspect Notebook Entry Network Requests', async ({ page }) => {
+        // Expand sidebar
+        await page.locator('.c-notebook__toggle-nav-button').click();
+
+        // Collect all request events to count and assert after notebook action
+        let requests = [];
+        page.on('request', (request) => requests.push(request));
+
+        let [notebookUrlRequest, allDocsRequest] = await Promise.all([
+            // Waits for the next request with the specified url
+            page.waitForRequest(`**/openmct/${testNotebook.uuid}`),
+            page.waitForRequest('**/openmct/_all_docs?include_docs=true'),
+            // Triggers the request
+            page.click('text=Page Add >> button'),
+            // Ensures that there are no other network requests
+            page.waitForLoadState('networkidle')
+        ]);
+        // Assert that only two requests are made
+        expect(requests.length).toBe(2);
+
+        // Assert on request object
+        expect(notebookUrlRequest.postDataJSON().metadata.name).toBe('TestNotebook');
+        expect(notebookUrlRequest.postDataJSON().model.persisted).toBeGreaterThanOrEqual(notebookUrlRequest.postDataJSON().model.modified);
+        expect(allDocsRequest.postDataJSON().keys).toContain(testNotebook.uuid);
+
+        // Add an entry
+        requests = [];
+        await page.locator('text=To start a new entry, click here or drag and drop any object').click();
+        await page.locator('[aria-label="Notebook Entry Input"]').click();
+        await page.locator('[aria-label="Notebook Entry Input"]').fill(`First Entry`);
+        await page.waitForLoadState('networkidle');
+        expect(requests.length).toBeLessThanOrEqual(2);
+
+        // Add some tags
+        requests = [];
+        await page.hover(`button:has-text("Add Tag")`);
+        await page.locator(`button:has-text("Add Tag")`).click();
+        await page.locator('[placeholder="Type to select tag"]').click();
+        await page.locator('[aria-label="Autocomplete Options"] >> text=Driving').click();
+        await page.waitForSelector('.c-tag__label:has-text("Driving")');
+        page.waitForLoadState('networkidle');
+        expect(requests.length).toBeLessThanOrEqual(12);
```

Comment:
How do we arrive at these numbers? Maybe we should add in some comments throughout indicating how many requests should be generated by a certain action.

Reply:
That's a good idea